### PR TITLE
Added week1 homework tests and updated documentation

### PR DESCRIPTION
**Pull request by GeorgeSapkin**
Might not be the best way to do it so work in progress.

### CODE REVIEW

**Comment by remarcmij** [comment]
I will give your homework week 1 a try using the instructions you provided and then give you feedback.

File: week1/homework/README.md [hunk 1]
```diff
@@ -1,25 +1,90 @@
...
+, /subtract returns 0 (10ms)
+    ✓ querying undefined URL returns 404 Not Found (1ms)
+```
+
+## Run the project
+
+In `week1/homework` run:
 
 ```bash
 node .
```
Comment (remarcmij) [inline week1/homework/README.md]: To clarify between using `node .` and running the tests:

You can interactively test the endpoints with the browser or by using PostMan.

Before submitting your homework via pull request, make sure that your server passes all the unit tests as describe above.

File: week1/homework/README.md [hunk 2]
```diff
@@ -28,76 +93,68 @@ node .
 ## Assignment
 
 Create an HTTP server that can add and subtract from a number, which we will
-call `state`. Use project in `week1` directory as starting material.
-Pay extra attention to line 21, which contains some hints for this week
-`console.log('New http request received', request.url);`
+call `state`. Use project in `week1/lecture` directory for reference.
+
```
Comment (remarcmij) [inline week1/homework/README.md]: Your modifications should be limited to the `src` folder. Do not make any changes to the `test` folder.

File: week1/homework/test/server.test.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        server.close();
+        throw err;
+      }
+    server.close();
+
+    const parsed = JSON.parse(response);
+
+    expect(parsed).toEqual({
+      state
+    });
```
Comment (remarcmij) [inline week1/homework/test/server.test.js]: Do you also want to test for `Content-Type application/json` in the response header?

File: week1/homework/src/index.js [hunk 4]
```diff
@@ -1,3 +1,9 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const {
+  createServer
+} = require('./server');
+
+const PORT = 3000;
+
+createServer(PORT);
```
Comment (remarcmij) [inline week1/homework/src/index.js]: I found it useful to have some feedback when the server is started:

```js
createServer(PORT)
  .then(() => {
    console.log('server listening on port', PORT);
  });
```

File: week1/homework/README.md [hunk 5]
```diff
@@ -28,76 +93,68 @@ node .
 ## Assignment
 
 Create an HTTP server that can add and subtract from a number, which we will
-call `state`. Use project in `week1` directory as starting material.
-Pay extra attention to line 21, which contains some hints for this week
-`console.log('New http request received', request.url);`
+call `state`. Use project in `week1/lecture` directory for reference.
```
Comment (remarcmij) [inline week1/homework/README.md]: Use **_the_** project...

File: week1/homework/README.md (hunk 2 above)
Comment (GeorgeSapkin) [reply week1/homework/README.md]: Make sense. Should have added that.

File: week1/homework/test/server.test.js (hunk 3 above)
Comment (GeorgeSapkin) [reply week1/homework/test/server.test.js]: Good idea.

**Comment by remarcmij** [comment]
Re-ran the tests on my "homework", forcing errors incl. content-type. All fine! Good work!

**Pull request merged**